### PR TITLE
table nested in drawer

### PR DESCRIPTION
**Pull request by danethurber**
fixes: #460

- style full tables that are nested inside a Drawer
- some refactoring of the table component to bring up to current standards

TODO:

- [x] docs

### CODE REVIEW

File: packages/table/src/css/index.js [hunk 1]
```diff
@@ -7,22 +7,17 @@ import { vars as drawerVars } from '@pluralsight/ps-design-system-drawer'
...
-}
-
 export default {
   '.psds-table': {
     width: '100%',
     fontWeight: core.type.fontWeightBook,
     fontSize: core.type.fontSizeSmall,
     lineHeight: core.type.lineHeightTight
   },
+  '.psds-table--in-drawer': {
```
Comment (danethurber) [inline packages/table/src/css/index.js]: this is new

File: packages/table/src/react/__stories__/index.story.js [hunk 2]
```diff
@@ -420,3 +414,36 @@ const drawerStory = storiesOf('drawer', module)
...
+        <Table.Cell>Parent Cell</Table.Cell>
+        <Table.Cell>Parent Cell</Table.Cell>
+        <Table.Cell>Parent Cell</Table.Cell>
+      </Table.Row>
+    )
+
+    return (
+      <Table>
+        <Drawer base={baseRow} startOpen>
+          <Table inDrawer>
```
Comment (danethurber) [inline packages/table/src/react/__stories__/index.story.js]: added a new `inDrawer` prop to add styling. alternative is Drawer could check if child is a table at add prop using cloneElement

Comment (jaketrent) [reply packages/table/src/react/__stories__/index.story.js]: A few thoughts:

- I want to minimize whatever any component knows about the other.  
- I want to minimize the little "mode" tweaks that devs have to add to props
- I want to minimize total props on components.

In this case, the first point bears the most weight for me in general.  

The Table already knows about Drawer (it horz pads its own rows if it has Drawer children, which have horz padd as well).  

I think the `inDrawer` prop is probably as good as the known alternate solutions.

So, ^ that's just me talking through it. :)

File: packages/table/src/react/index.js [hunk 3]
```diff
@@ -1,71 +1,45 @@
-import { vars as drawerVars } from '@pluralsight/ps-design-system-drawer'
 import * as glamor from 'glamor'
-import Icon from '@pluralsight/ps-design-system-icon/react'
-import { defaultName as themeDefaultName } from '@pluralsight/ps-design-system-theme/react'
 import PropTypes from 'prop-types'
 import React from 'react'
 
+import { vars as drawerVars } from '@pluralsight/ps-design-system-drawer'
```
Comment (jaketrent) [inline packages/table/src/react/index.js]: I would use `/vars` in the import instead in order to not get the whole Drawer code included here. (current untreeshakeable)

File: packages/table/src/react/index.js [hunk 4]
```diff
@@ -128,17 +104,22 @@ class ColumnHeader extends React.Component {
     )
   }
 }
+
+ColumnHeader.displayName = 'Table.ColumnHeader'
+
+/* eslint-disable react/no-unused-prop-types */
```
[the remark this answers is not in the document]
Comment (danethurber) [reply packages/table/src/react/index.js]: the linter can't handle it. the props are passed to a style function and the linter can't follow along

**Pull request merged**